src/main/scala/CommandHandlers.scala

```diff
@@ -3,7 +3,9 @@ package simpleCQRS
 class InventoryCommandHandlers(repository: Repository[InventoryItem]) {
 
   def handle(cmd: CreateInventoryItem) {
-    repository.save(new InventoryItem(cmd.inventoryItemId, cmd.name), -1);
+    val item = new InventoryItem(cmd.inventoryItemId, cmd.name)
+    item.init()
```

Comment:
Partial initialization in constructor and in init() breaks invariants of class. After constructor execution aggregate must be fully initialized and usable.